azurerm/resource_arm_sql_server_test.go

```diff
@@ -99,14 +100,16 @@ func testCheckAzureRMSqlServerDestroy(s *terraform.State) error {
 		sqlServerName := rs.Primary.Attributes["name"]
 		resourceGroup := rs.Primary.Attributes["resource_group_name"]
 
-		resp, err := conn.Get(resourceGroup, sqlServerName)
-
-		if err != nil {
-			return nil
-		}
-
-		if resp.Response.StatusCode != http.StatusNotFound {
-			return fmt.Errorf("SQL Server still exists:\n%#v", resp.ServerProperties)
+		_, err := conn.Get(resourceGroup, sqlServerName)
+		if autorestErr, ok := err.(autorest.DetailedError); ok {
+			switch autorestErr.StatusCode {
+			case http.StatusNotFound:
+				return nil
+			default:
+				return err
+			}
+		} else {
+			return fmt.Errorf("Bad type assertion for error returned from sqlServersClient.")
 		}
```

Comment:
As far as I know there's no guarantees given on Error types returned by the SDK - so I think this might be safer + simpler to instead read the status code from the Response object?  [Here's an example of a recently refactored test](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_dns_zone_test.go#L102) which shows what I mean - what do you think?

(an aside: I think this would read cleaner as an if statement anyway, given we're only concerned with a 404 here?)